fdbserver/worker.actor.cpp

```diff
@@ -2706,8 +2706,8 @@ ACTOR Future<Void> updateNewestSoftwareVersion(std::string folder,
 		    0600));
 
 		SWVersion swVersion(latestVersion, currentVersion, minCompatibleVersion);
-		auto s = swVersionValue(swVersion);
-		ErrorOr<Void> e = wait(errorOr(newVersionFile->write(s.toString().c_str(), s.size(), 0)));
+		state Standalone<StringRef> s = swVersionValue(swVersion);
+		ErrorOr<Void> e = wait(errorOr(newVersionFile->write(s.begin(), s.size(), 0)));
```

Comment:
Just a comment. We actually have `holdWhile` and `holdWhileVoid` actors to prevent object deconstruction.

Reply:
Yup, I think that would be an alternative to making `s` a state variable. I don't have a preference. @sfc-gh-xwang are you saying you prefer `holdWhile` to `s` being a state variable? Happy to make that change if so

Reply:
If you do keep it a state variable, I would recommend renaming it. We generally try to discourage short names for state variables because of shadowing risks.

Reply:
Yes. I prefer `holdWhile`. I just feel using state for a temp variables is somewhat too much, they don't need a lifetime as long as the outside actor.